packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -480,20 +491,37 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
 
     final List<MergeableMaterialItem> items = <MergeableMaterialItem>[];
 
+    
+
     for (int index = 0; index < widget.children.length; index += 1) {
       if (_isChildExpanded(index) && index != 0 && !_isChildExpanded(index - 1))
         items.add(MaterialGap(key: _SaltedKey<BuildContext, int>(context, index * 2 - 1)));
-
       final ExpansionPanel child = widget.children[index];
       final Widget headerWidget = child.headerBuilder(
         context,
         _isChildExpanded(index),
       );
 
+      final bool _isExpanded = _isChildExpanded(index);
+      final Set<MaterialState> states = <MaterialState>{};
+
+      if (_isExpanded && !child.canTapOnHeader) {
+        states.add(MaterialState.selected);
+      }
+
+      if (child.canTapOnHeader) {
+        states.add(MaterialState.disabled);
+      }
+
+      final Color? _expandIconColor = child.iconColor?.resolve(states);
+
```

Comment:
Do you think this is the correct approach?

Reply:
Yes, it feels like this is on the right path. I think we're getting close.